exporter/trace/trace_proto_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"testing"
+
+	export "go.opentelemetry.io/otel/sdk/export/trace"
+	"go.opentelemetry.io/otel/sdk/resource"
+
+	"go.opentelemetry.io/otel/api/kv"
+)
+
+func TestInjectLabelsFromResources(t *testing.T) {
+	testcases := []struct {
+		input    export.SpanData
+		expected export.SpanData
+	}{
+		// empty resource
+		{
+			input: export.SpanData{
+				Resource: resource.New(),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+				},
+			},
+			expected: export.SpanData{
+				Resource: resource.New(),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+				},
+			},
+		},
+		// empty attributes
+		{
+			input: export.SpanData{
+				Resource:   resource.New(kv.String("b", "2")),
+				Attributes: []kv.KeyValue{},
+			},
+			expected: export.SpanData{
+				Resource: resource.New(kv.String("b", "2")),
+				Attributes: []kv.KeyValue{
+					kv.String("b", "2"),
+				},
+			},
+		},
+		// normal insert
+		{
+			input: export.SpanData{
+				Resource: resource.New(kv.String("b", "2")),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+				},
+			},
+			expected: export.SpanData{
+				Resource: resource.New(kv.String("b", "2")),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+					kv.String("b", "2"),
+				},
+			},
+		},
+		// conflicts with the existing keys
+		{
+			input: export.SpanData{
+				Resource: resource.New(kv.String("a", "2")),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+				},
+			},
+			expected: export.SpanData{
+				Resource: resource.New(kv.String("a", "2")),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+				},
+			},
+		},
+		// duplicate keys in attributes (allowed)
+		{
+			input: export.SpanData{
+				Resource: resource.New(
+					kv.String("c", "1"),
+				),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+					kv.String("b", "1"),
+					kv.String("b", "2"),
+					kv.String("b", "3"),
+				},
+			},
+			expected: export.SpanData{
+				Resource: resource.New(kv.String("c", "1")),
+				Attributes: []kv.KeyValue{
+					kv.String("a", "1"),
+					kv.String("b", "1"),
+					kv.String("b", "2"),
+					kv.String("b", "3"),
+					kv.String("c", "1"),
+				},
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.input.Resource.String(), func(t *testing.T) {
```

Comment:
It's a bit weird to make the resource string the test case name here. A couple of your tests have the same resource so presumably the same test case name.

It would be better to make your comment for each case the name instead, i.e.

```go
testcases := []struct {
	name     string
	input    export.SpanData
	expected export.SpanData
}{
	{
		name: empty resource,
		input: ...
...
}

...

t.Run(tc.name, ...
```